File: backend/src/main/java/com/devconnect/models/Post.java
package com.devconnect.models;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Data;

import java.time.Instant;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.UUID;

@Data
public class Post {
    private UUID id = UUID.randomUUID();
    
    @NotBlank
    @Size(min = 1, max = 255)
    private String title;
    
    private String content;
    
    private List<CodeBlock> codeBlocks = new ArrayList<>();
    
    @NotBlank
    private UUID authorId;
    
    @NotBlank
    @Size(min = 3, max = 30)
    private String authorAlias;
    
    private Instant createdAt = Instant.now();
    
    private int commentCount;
    
    private Map<ReactionType, Integer> reactions = new HashMap<>();
    
    public enum ReactionType {
        HEART, THUMBS_UP, ROCKET
    }
}

File: android/app/src/main/kotlin/com/devconnect/data/model/Post.kt
package com.devconnect.data.model

import java.time.Instant
import java.util.UUID

data class Post(
    val id: UUID,
    val title: String,
    val content: String?,
    val codeBlocks: List<CodeBlock>,
    val authorId: UUID,
    val authorAlias: String,
    val createdAt: Instant,
    val commentCount: Int,
    val reactions: Map<ReactionType, Int>
) {
    enum class ReactionType {
        HEART, THUMBS_UP, ROCKET
    }
}

data class CodeBlock(
    val content: String,
    val language: String,
    val position: Int,
    val foldingRanges: List<FoldingRange> = emptyList()
) {
    data class FoldingRange(
        val startLine: Int,
        val endLine: Int,
        val kind: String,
        val label: String?
    )
}
